packages/sdk/src/client/edition/mint.ts

```diff
@@ -131,8 +139,145 @@ export async function mint(
+    maxFeePerGas,
+    maxPriorityFeePerGas,
+    mintSchedule,
+    mintToAddress,
+    quantity,
+  }: MintToOptions,
+): Promise<ContractTransaction> {
+  if (!isBigNumberish(attributonId)) {
+    throw new InvalidAttributonIdError({
+      attributonId,
+    })
+  }
+
+  await validateSoundEdition.call(this, { editionAddress: mintSchedule.editionAddress })
+  if (quantity <= 0 || Math.floor(quantity) !== quantity) throw new InvalidQuantityError({ quantity })
+
+  const { signer, userAddress } = await this.expectSigner()
+
+  const toAddress = mintToAddress ?? userAddress
+
+  const eligibleMintQuantity = await eligibleQuantity.call(this, {
+    mintSchedule,
+    userAddress,
+  })
+  if (eligibleMintQuantity < quantity) {
+    throw new NotEligibleMint({
+      eligibleMintQuantity,
+      mintSchedule,
+      userAddress,
+    })
+  }
+
+  const txnOverrides: PayableOverrides = {
+    value: 'price' in mintSchedule ? mintSchedule.price.mul(quantity) : BigNumber.from('0'),
+    gasLimit,
+    maxFeePerGas,
+    maxPriorityFeePerGas,
+  }
+
+  const interfaceId = mintSchedule.interfaceId
+
+  switch (interfaceId) {
+    case interfaceIds.IRangeEditionMinterV2: {
+      const rangeMinter = minterFactoryMap[interfaceId].connect(mintSchedule.minterAddress, signer)
+
+      const mintArgs = [
+        mintSchedule.editionAddress,
+        mintSchedule.mintId,
+        toAddress,
+        quantity,
+        affiliate,
+        affiliateProof,
+        attributonId,
+      ] as const
+
+      if (txnOverrides.gasLimit) {
+        return rangeMinter.mintTo(...mintArgs, txnOverrides)
+      }
+
+      try {
+        // Add a buffer to the gas estimate to account for node provider estimate variance.
+        const gasEstimate = await rangeMinter.estimateGas.mintTo(...mintArgs, txnOverrides)
+
+        txnOverrides.gasLimit = scaleAmount({ amount: gasEstimate, multiplier: MINT_GAS_LIMIT_MULTIPLIER })
+      } catch (err) {
+        // If estimation fails, provide a hardcoded gas limit that is guaranteed to succeed.
+        txnOverrides.gasLimit = MINT_FALLBACK_GAS_LIMIT
+      }
+
+      return rangeMinter.mintTo(...mintArgs, txnOverrides)
+    }
+
+    case interfaceIds.IMerkleDropMinterV2: {
+      const merkleDropMinter = minterFactoryMap[interfaceId].connect(mintSchedule.minterAddress, signer)
+
+      const { merkleRootHash: merkleRoot } = await merkleDropMinter.mintInfo(
+        mintSchedule.editionAddress,
+        mintSchedule.mintId,
+      )
+
+      const proof = await getMerkleProof.call(this, {
+        merkleRoot,
+        userAddress,
+      })
+
+      if (!proof?.length) {
+        throw new NotEligibleMint({
+          mintSchedule,
+          userAddress,
+          eligibleMintQuantity,
+        })
+      }
+
+      const mintArgs = [
+        mintSchedule.editionAddress,
+        mintSchedule.mintId,
+        toAddress,
+        quantity,
+        // TODO, allow overriding allowlisted for delegatecash
+        toAddress,
```

Comment:
at contract level, we allow one address to mint on behalf of a different whitelisted address
 if it has been registered on delegate cash 
 
 this is not something we plan to expose on initial launch, but may circle back to fixing sdk to take that extra param